### PR TITLE
Added default Nomad query options and set AllowStale to true Govet fix for AWS autoscaling package

### PR DESCRIPTION
**Pull request by vladshub**
This is a suggested implementation for https://github.com/elsevier-core-engineering/replicator/issues/224

### CODE REVIEW

**Comment by vladshub** [comment]
@ericwestfall, @jrasell It seems that I was a few minuets too late https://github.com/elsevier-core-engineering/replicator/pull/227 please decide which PR you like :+1:

**Comment by jrasell** [comment]
Hey @vladshub and thanks for another PR. I have read through the code changes and the description and think I am going to close this PR without merging for a couple of reasons. The first is that the PR addresses two very different changes, and so should be made through two distinct pull requests. 

The second is that the AWS autoscaling package is named purposefully, and as such, vet should be ignored in this instance. The functions are AWS specific rather than cloud generic and so need to be differentiated from generic cloud functions. I won't close this straight away to allow for addition comments.

**Comment by vladshub** [comment]
@jrasell thanks for the review... As I said only after I pushed it I saw that you pushed a PR that you have made... 
Regarding changes in the API requests to nomad I think it was a good idea to have all of the settings in one place in case we would need to have more common params...

**Comment by vladshub** [comment]
Feel free to close this when ever you like :)

File: client/nomad.go [hunk 1]
```diff
@@ -366,6 +366,11 @@ func (c *nomadClient) GetAllocationStats(allocation *nomad.Allocation, scalingPo
 			scalingPolicy.Tasks.Resources.MemoryMB)
 }
 
+// queryOptions returns a copy of default nomad query options set in NewNomadClient
```
Comment (ericwestfall) [inline client/nomad.go]: This is a nice approach to setting the query options and I would like to see this carried over to #224

**Comment by ericwestfall** [comment]
Thanks for the PR @vladshub, as I stated in my comment above, I really like the approach you took to setting the query options and we'll see about using that in #227. 

Regarding the Cloud Provider rename, we are going to restructure that a bit and do some renaming so the `vet` warnings should go away. That work is being done under #216.

Sorry the timing didn't work out on this PR!

**Pull request closed without merging**